torch/csrc/api/include/torch/nn/modules/any.h

```diff
@@ -315,6 +322,16 @@ template <typename ModuleType>
 AnyModule::AnyModule(const ModuleHolder<ModuleType>& module_holder)
     : AnyModule(module_holder.get()) {}
 
+inline AnyModule::AnyModule(const AnyModule& other)
+    : content_(other.content_ ? other.content_->clone() : nullptr) {}
```

Comment:
I'm skeptical about this. Is this doing a deep clone? I don't expect a deep clone to happen unless I explicitly ask for it using `clone()`.

Reply:
It makes a copy of the `AnyModule`, but this in turn makes a copy of the `shared_ptr<Module>` it has inside. So it's more like a shallow copy. I agree that to be consistent with the rest of the library, deep copies should be explicit.

Reply:
OK, so then the obvious follow up question is, why doesn't clone on AnyModule do a deep clone? :)

Reply:
I see what you mean, but I didn't add `AnyModule::clone()`, so it can't be called or anything. You'd have to access the contained module, and clone it. If someone asks for it, I could add `AnyModule::clone()` which would then do a deep-copy. That would be consistent with everything else, e.g. `Tensor::clone()` does a deep copy and `Tensor`'s copy constructor does a shallow copy. At the moment, there's not yet a need for `AnyModule::clone()`.